internal/mode/static/usage/secret.go

```diff
@@ -0,0 +1,60 @@
+package usage
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . SecretStore
+
+// SecretStore should store the usage Secret that contains the credentials for NGINX Plus usage reporting.
+type SecretStore interface {
```

Comment:
Can we define this interface on the client-side? Also, interface names are typically verbs like `Reader`. What about naming this `Storer` or `Setter`?

Reply:
Where on the client side? You mean in the `static` package where manager and handler live?

Reply:
That's what I was thinking initially, but I overlooked the fact that the reporter also uses this interface. 

Go recommends declaring the interface where it is used. In this case, it is used in the `static` and `usage` packages. Historically, we've been bad at following this best practice.

After doing a bit of reading about this, the recommended approach is to declare the interface twice, once in each package where it is used. With this solution, you won't need to export the interfaces. This will be a little bit repetitive, but it keeps each package self-contained. 

@pleshakov curious what your thoughts are on this.

Reply:
I think it's better to define it client side.
This way, in the Reported, we only need to use GetCredentials, but in the Handler, we would only care about Set and Delete.

Reply:
Good point, @pleshakov. That makes defining two interfaces a more palatable option.